README.md

```diff
@@ -1,2 +1,232 @@
+  |                                    .                .
+  +------------>  Service------------- . ----------+    .
+  |                                    .           |    .
+  +------------->  Protocol----------- . ----------+    .
+  |                                    .           |    .
+  +-------------->* Peer               \/          |    .
+                     +------------> Connection     |    .
+                     |                             \/   \/
+                     +--------------------------->  Stream
+```
+
+### The System Scope
+
+The system scope is the top level scope that accounts for global
+resource usage at all levels of the system. This scope constrains all
+other scopes and institutes global hard limits.
+
+### The Transient Scope
+
+The transient scope accounts for resources that are in the process of
+full establishment.  For instance, a new connection prior to the
+handshake does not belong to any peer, but it still needs to be
+constrained as this opens an avenue for attacks in transient resource
+usage. Similarly, a stream that has not negotiated a protocol yet is
+constrained by the transient scope.
+
+### Service Scopes
+
+The system is typically organized across services, which may be
+ambient and provide basic functionality to the system (e.g. identify,
+autonat, relay, etc). Alternatively, services may be explicitly
+instantiated by the application, and provide core components of its
+functionality (e.g. pubsub, the DHT, etc).
+
+Services consume resources such as memory and may directly own streams
+that implement their protocol flow. Services typically have some
+stream handler, so they are subject to inbound stream creation and
+resource usage in response to network events. As such, the system
+explicitly models them allowing for isolated resource usage that can
+be tuned by the user.
+
+### Protocol Scopes
+
+Protocol Scopes account for resources at the protocol level. They are
+an intermediate resource scope which can constrain streams which may
+not have a service associated or for resource control within a
+service.
+
+For instance, a service that is not aware of the resource manager and
+has not be ported to mark its streams, may still gain limits
+transparently without any programmer intervention.  Furthermore, the
+protocol scope can constrain resource usage for services that
+implement multiple protocols for the shake of backwards
+compatibility. A tighter limit in some older protocol can protect the
+application from resource consumption caused by legacy clients or
+potential attacks.
+
+For a concrete example, consider pubsub with the gossipsub router: the
+service also understands the floodsub protocol for backwards
+compatibility and support for unsophisticated clients that are lagging
+in the implementation effort. By specifying a lower limit for the
+floodsub protocol, we can can constrain the service level for legacy
+clients using an inefficient protocol.
+
+### Peer Scopes
```

Comment:
What do we allow specifying with peer scopes:
1. No single peer may use more than X resources
2. Explicit peer Y may not use more X resources
3. Both
?

Reply:
Both actually. We first lookup if a limit has been specified for a peer and if not we use the default peer limits.
Will add some text to clarify.